### PR TITLE
[tests-only] added unit test for oc avatar component

### PR DESCRIPTION
**Pull request by kiranparajuli589**
## Description
With this PR: 
- unit tests for `OcAvatar` component are added

## Related Issue

- part of #1366 

## Motivation and Context
- increase unit test coverage

## How Has This Been Tested?

- 🤖 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

## Checklist:

- [ ] Code changes
- [x] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

## Open tasks:

- [ ] ...

### CODE REVIEW

File: src/components/avatars/OcAvatar.spec.js [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        ${"Test User"}             | ${"TU"}
+        ${"1234 5678"}             | ${"15"}
+        ${"test-user"}             | ${"TU"}
+        ${"test -user"}            | ${"TU"}
+        ${"test - user"}           | ${"TU"}
+        ${"test user one"}         | ${"TUO"}
+        ${"test-user-one"}         | ${"TUO"}
+        ${"test user one primary"} | ${"TUO"}
+        ${"test User One Primary"} | ${"UOP"}
+        ${"testUser One Primary"}  | ${"TOP"}
```
Comment (individual-it) [inline src/components/avatars/OcAvatar.spec.js:52]: lets try to have some strings that need UTF8 e.g.  किरन

File: src/components/avatars/OcAvatar.spec.js [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+        })
+      })
+      it("should set width and height of the avatar wrapper", () => {
+        expect(wrapper.attributes("style")).toContain("width: 100px; height: 100px;")
+      })
+      it("should determine font style", async () => {
+        expect(wrapper.attributes("style")).toContain("line-height: 105px;")
+        expect(wrapper.attributes("style")).toContain(
+          "font-size: 40px; font-family: Helvetica, Arial, sans-serif;"
+        )
```
Comment (individual-it) [inline src/components/avatars/OcAvatar.spec.js:76]: that feels like testing too much implementation details

File: src/components/avatars/OcAvatar.spec.js [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+      `("should render user initials", ({ username, expected_initial }) => {
+        const wrapper = getWrapperWithProps({
+          userName: username,
+        })
+        const userInitialElement = wrapper.find(selectors.initials)
+        expect(userInitialElement.exists()).toBeTruthy()
+        expect(userInitialElement.text()).toBe(expected_initial)
+      })
+    })
+    describe("width", () => {
```
Comment (individual-it) [inline src/components/avatars/OcAvatar.spec.js:62]: use similar describe everywhere on this level
"when username is set" vs "width"

**Review by individual-it — CHANGES_REQUESTED** [review]

File: src/components/avatars/OcAvatar.spec.js (hunk 2 above)
Comment (kiranparajuli589) [reply src/components/avatars/OcAvatar.spec.js:76]: styles are calculated using `height` inside the component, so I added the tests for it too.

File: src/components/avatars/OcAvatar.spec.js (hunk 1 above)
Comment (kiranparajuli589) [reply src/components/avatars/OcAvatar.spec.js:59]: added

File: src/components/avatars/OcAvatar.spec.js (hunk 3 above)
Comment (kiranparajuli589) [reply src/components/avatars/OcAvatar.spec.js]: fixed

File: src/components/avatars/OcAvatar.spec.js [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+      })
+      it("should not render background", () => {
+        expect(wrapper.attributes("style")).not.toContain("background-color: ")
+      })
+    })
+    describe("when username is set", () => {
+      it.each`
+        username                   | expected_initial
+        ${"."}                     | ${"."}
+        ${"ॐ 卐"}                  | ${"ॐ卐"}
```
Comment (kulmann) [inline src/components/avatars/OcAvatar.spec.js:39]: I know that it has different meaning for you, but please avoid having a svastika. 😅

File: src/components/avatars/OcAvatar.spec.js (hunk 2 above)
Comment (individual-it) [reply src/components/avatars/OcAvatar.spec.js]: Good point, you are right :+1: 
lets then reduce it to the calculated values and leave `font-family` alone          `expect(wrapper.attributes("style")).toContain("font-size: 40px")`

File: src/components/avatars/OcAvatar.spec.js (hunk 4 above)
Comment (kiranparajuli589) [reply src/components/avatars/OcAvatar.spec.js]: removed ;) !

**Review by individual-it — APPROVED** [review]

**Pull request merged**